packages/curve-ui-kit/src/shared/ui/SwipeableDrawer.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import { Box, styled } from '@mui/material'
+import MuiSwipeableDrawer from '@mui/material/SwipeableDrawer'
+type Props = {
+  button: React.ReactNode
+  children: React.ReactNode
+  open: boolean
+  setOpen: (open: boolean) => void
+}
+
+/**
+ * The following properties are advised for optimal usability: https://mui.com/material-ui/react-drawer/#swipeable
+ * iOS is hosted on high-end devices. The backdrop transition can be enabled without dropping frames. The performance will be good enough.
+ * iOS has a "swipe to go back" feature that interferes with the discovery feature, so discovery has to be disabled.
+ */
+const iOS = typeof navigator !== 'undefined' && /iPad|iPhone|iPod/.test(navigator.userAgent)
+
+/**
+ * A swipeable drawer component that can be used to display a list of items.
+ * @param param0 button: React.ReactNode, children: React.ReactNode, open: boolean, setOpen: (open: boolean) => void
+ */
+export const SwipeableDrawer = ({ button, children, open, setOpen }: Props) => (
+  <Box sx={{ height: '100%' }}>
+    {button}
+    <MuiSwipeableDrawer
+      disableBackdropTransition={!iOS}
+      disableDiscovery={iOS}
+      anchor="bottom"
+      open={open}
+      onClose={() => setOpen(false)}
+      onOpen={() => setOpen(true)}
+      disableSwipeToOpen={false}
+      keepMounted={false}
+    >
+      <Box sx={{ pt: 2 }}>
```

Comment:
Let's avoid acronyms please. Also we are trying to use the block/inline versions (instead of x/y) that are supposed to be the best practice

```suggestion
      <Box sx={{ paddingBlockStart: 2 }}>
```